Gems/LyShine/Code/Tests/AnimationTest.cpp

```diff
@@ -81,6 +88,20 @@ namespace UnitTest
             m_timeSystem = AZStd::make_unique<UnitTest::AnimationTestStubTimer>();
         }
 
+        // override and only include system components required for tests.
+        AZ::ComponentTypeList GetRequiredSystemComponents() const override
```

Comment:
Rather than excluding the UserSettingsComponent this way, you can also send a 
 'DisableSaveOnFinalize' broadcast call to prevent writing to file. This is a pretty common thing done in test fixtures across the engine. [Here is an example](https://github.com/o3de/o3de/blob/development/Code/Framework/AzToolsFramework/Tests/TransformComponent.cpp#L52-L55) . 
 
Doing so will allow the tests to mirror actual entities, which might be useful in catching errors with other components.

Reply:
> Rather than excluding the UserSettingsComponent this way, you can also send a 'DisableSaveOnFinalize' broadcast call to prevent writing to file. This is a pretty common thing done in test fixtures across the engine. [Here is an example](https://github.com/o3de/o3de/blob/development/Code/Framework/AzToolsFramework/Tests/TransformComponent.cpp#L52-L55) .
> 
> Doing so will allow the tests to mirror actual entities, which might be useful in catching errors with other components.

That was my initial fix but after finding the root cause I removed the DisableSaveOnFinalize call since it is no longer necessary. The UserSettingsComponentRequestBus isn't connected anymore.

Reply:
Well explicitly adding the components, adds a hidden dependency to maintainers of the `AzFramework::Application` class to need to update this file, when they change the list of required components.

Reply:
hmm that's a good point, but in that case why is GetRequiredSystemComponents() a virtual function in the first place? :)

Reply:
The primarily use case is to add more required components and then delegate to the base class function.